src/main/scala/com/codacy/parsers/implementation/CloverParser.scala

```diff
@@ -43,9 +44,23 @@ object CloverParser extends CoverageParser with XmlReportParser {
     MathUtils.computePercentage(coveredStatements, totalStatements)
   }
 
-  private def getCoverageFileReport(rootPath: String, fileNode: Node) = {
-    val filename = fileNode \@ "name"
-    val cleanFileName = TextUtils.sanitiseFilename(filename).stripPrefix(rootPath).stripPrefix("/")
+  private def getCoverageFileReport(rootPath: String, fileNode: Node): CoverageFileReport = {
+    val filePath = fileNode.attribute("path").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+    val filename = fileNode.attribute("name").flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
+
+    val relativeFilePath = filePath
+      .orElse(filename)
+      .fold[String] {
+        throw new Exception(
+          "Could not read file path due to missing 'path' and 'name' attributes in the report file element."
```

Comment:
Minor nitpick: the message should actually be "(...) due to missing 'path' **or** 'name' attributes (...)"

Reply:
That is a good question. I had or initially, but changed.
We just need one of them, so I want to say they both were missing. Doesn't and make more sense?

Reply:
If you only need on, then `or` definitely makes more sense to me. Anyway, this is just a nitpick, so I'll approve anyway.